adaptors/metal3/baremetalhost_manager.go

```
@@ -505,6 +507,41 @@ func (a *Adaptor) handleTransitionNodes(ctx context.Context, nodelist *hwmgmtv1a
 	return false, nil
 }
 
+func (a *Adaptor) addRebootAnnotation(ctx context.Context, bmh *metal3v1alpha1.BareMetalHost) error {
+	bmhName := types.NamespacedName{Name: bmh.Name, Namespace: bmh.Namespace}
+	if err := a.updateBMHMetaWithRetry(ctx, bmhName, "annotation", BmhRebootAnnotation, "", OpAdd); err != nil {
+		return fmt.Errorf("failed to add %s to BMH %+v: %w", BmhRebootAnnotation, bmhName, err)
+	}
+	return nil
+}
+
+func (a *Adaptor) evaluateCRForReboot(ctx context.Context, bmh *metal3v1alpha1.BareMetalHost, annotation string) error {
+
+	var changeDetected bool
+	var err error
+
+	switch annotation {
+	case BiosUpdateNeededAnnotation:
+		changeDetected, err = a.isFirmwareSettingsChangeDetected(ctx, bmh)
+		if err != nil {
+			return fmt.Errorf("failed to evalute FirmwareSettings status: %w", err)
+		}
+
+	case FirmwareUpdateNeededAnnotation:
+		changeDetected, err = a.isHostFirmwareComponentsChangeDetected(ctx, bmh)
+		if err != nil {
+			return fmt.Errorf("failed to evalute HostFirmwareComponents status: %w", err)
+		}
+	default:
+		return fmt.Errorf("unsupported: %s", annotation)
```

Do we need to return an error for an unsupported annotation? Why not log a message `unsupported annotation...` and `return nil` here?

---

Yes, updated.